lighthouse-core/gather/driver.js

```diff
@@ -428,6 +428,23 @@ class Driver {
   }
 
   /**
+   * @param {string} name The name of API whose permission you wish to query
+   * @return {!Promise<string>} The state of permissions, resolved in a promise.
+   *    See https://developer.mozilla.org/en-US/docs/Web/API/Permissions/query.
+   */
+  queryPermissionState(name) {
+    const expressionToEval = `
+      (function () {
```

Comment:
Eh I guess no wrap necessary, and just in case `name` contained a `'` or `"` didn't want an inscrutable error message